tests/integration/kubernetes/k8s-hostname.bats

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bats
+#
+# Copyright (c) 2024 Ant Group
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/../../common.bash"
+load "${BATS_TEST_DIRNAME}/tests_common.sh"
+
+setup() {
+	pod_name="test-pod-hostname"
+	get_pod_config_dir
+
+	yaml_file="${pod_config_dir}/pod-hostname.yaml"
+	add_allow_all_policy_to_yaml "${yaml_file}"
+
+	expected_name=$pod_name
+}
+
+@test "Validate Pod hostname" {
+	# Create pod
+	kubectl apply -f "${yaml_file}"
+
```

Comment:
Unlike what I previously suggested in https://github.com/kata-containers/kata-containers/pull/10423#discussion_r1804380701, we still need a logic to wait for the POD to be ready otherwise `kubectl logs` also can fail :

```
# Error from server (BadRequest): container "pod-hostname-c01" in pod "test-pod-hostname" is waiting to start: ContainerCreating
```

Since what we actually want is the result of  `hostname`, let's wait for the pod to complete in a reasonable time :
```
kubectl wait --for jsonpath=status.phase=Succeeded pods pod01-hostname --timeout=30s
```

Reply:
> kubectl wait --for jsonpath=status.phase=Succeeded pods
>
"  restartPolicy: Never" needed to make it run into completed state

Reply:
> Unlike what I previously suggested in [#10423 (comment)](https://github.com/kata-containers/kata-containers/pull/10423#discussion_r1804380701), we still need a logic to wait for the POD to be ready otherwise `kubectl logs` also can fail :
> 
> ```
> # Error from server (BadRequest): container "pod-hostname-c01" in pod "test-pod-hostname" is waiting to start: ContainerCreating
> ```
> 
"  restartPolicy: Never" needed to make it run into completed state

Reply:
```shell
# kubectl wait --for jsonpath=status.phase=Succeeded --timeout=10s pod/pod01-hostname
pod/pod01-hostname condition met
```